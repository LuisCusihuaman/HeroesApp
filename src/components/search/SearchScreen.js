import React from "react";
import { heroes } from "../../data/heroes";
import { HeroCard } from "../heroes/HeroCard";
import { useForm } from "../../hooks/useForm";

export const SearchScreen = () => {
	const heroesFiltered = heroes;
	const [formValues, handleInputChange, reset] = useForm({
		searchText: "",
	});
	const { searchText } = formValues;
	const handleSearch = (e) => {
		e.preventDefault();
		console.log(searchText);
		reset();
	};
	return (
		<div>
			<h1>SearchScreen</h1>
			<hr />
			<div className="row">
				<div className="col-5">
					<h4>Search form</h4>
					<form onSubmit={handleSearch}>
						<input
							type="text"
							placeholder="Find You Hero"
							className="form-control"
							name="searchText"
							value={searchText}
							onChange={handleInputChange}
						/>
						<button
							type="submit"
							className="btn m-1 btn-block btn-outline-primary"
						>
							Search ...
						</button>
					</form>
				</div>
				<div className="col-7">
					<h4>Results</h4>
					<hr />
					{heroesFiltered.map((hero) => (
						<HeroCard key={hero.id} {...hero} />
					))}
				</div>
			</div>
		</div>
	);
};
