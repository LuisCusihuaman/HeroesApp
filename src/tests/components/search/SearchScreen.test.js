import React from "react";
import { mount } from "enzyme";
import { SearchScreen } from "../../../components/search/SearchScreen";
import { MemoryRouter, Route } from "react-router-dom";

describe("Pruebas en <SearchScreen/>", () => {
	test("debe de mostrarse correctactamente con valores por defecto", () => {
		const wrapper = mount(
			<MemoryRouter initialEntries={["/search"]}>
				<Route path="/search" component={SearchScreen} />
			</MemoryRouter>
		);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find(".alert-info").text().trim()).toBe("Search a hero");
	});
	test("debe de mostrar a batman y el input con el valor del queryString", () => {
		const wrapper = mount(
			<MemoryRouter initialEntries={["/search?q=batman"]}>
				<Route path="/search" component={SearchScreen} />
			</MemoryRouter>
		);
		expect(wrapper.find("input").prop("value")).toBe("batman");
		expect(wrapper).toMatchSnapshot();
	});
});
