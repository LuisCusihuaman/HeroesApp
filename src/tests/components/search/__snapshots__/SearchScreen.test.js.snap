// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchScreen/> debe de mostrar a batman y el input con el valor del queryString 1`] = `
<div>
  <h1>
    SearchScreen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search form
      </h4>
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find You Hero"
          type="text"
          value="batman"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search ...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters "
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Batman"
              className="card-img"
              src="./assets/heroes/dc-batman.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Batman
              </h5>
              <p
                className="card-text"
              >
                Bruce Wayne
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Detective Comics #27
                </small>
              </p>
              <a
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en <SearchScreen/> debe de mostrarse correctactamente con valores por defecto 1`] = `
<div>
  <h1>
    SearchScreen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search form
      </h4>
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find You Hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search ...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
